safe/reportv4/extractors/action_notes.py

```diff
@@ -0,0 +1,80 @@
+# coding=utf-8
+import os
+
+from safe.definitionsv4.exposure import exposure_population, exposure_road, \
+    exposure_all
+from safe.utilities.i18n import tr
+from safe.definitionsv4.hazard_classifications import generic_hazard_classes
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+def action_checklist_extractor(impact_report, component_metadata):
+    """
+    Extracting action checklist of the exposure layer
+
+    :param impact_report: the impact report that acts as a proxy to fetch
+        all the data that extractor needed
+    :type impact_report: safe.reportv4.impact_report.ImpactReport
+    :param component_metadata: the component metadata. Used to obtain
+        information about the component we want to render
+    :type component_metadata: safe.reportv4.report_metadata.ReportMetadata
+
+    :return: context for rendering phase
+    :rtype: dict
+    """
+    context = {}
+
+    # figure out analysis report type
+    impact_layer = impact_report.impact_layer
+    """:type: qgis.core.QgsVectorLayer"""
```

Comment:
I used this for pycharm type hinting, really useful feature.
You will find some of these. Originally, I put this one since I code it backwards when 
impact_report still empty. But I deleted this one when the type in impact_report 
is fixed.
I'll delete any other if it is redundant with property type hinting and keep the rest where 
it is useful (such as variable in a loop).